modules/videoModule/index.js

```diff
@@ -0,0 +1,172 @@
+import { coreAdServerFactory } from '../adServerModule/adServer.js';
+import { videoImpressionVerifierFactory } from './videoImpressionVerifier.js';
+import { mergeDeep } from '../../src/utils.js';
+import { getGlobal } from '../../src/prebidGlobal.js';
+
+/**
+ * This module adds User Video support to prebid.js
+ * @module modules/videoModule
+ */
+
+events.addEvents(allVideoEvents);
+events.addEvents(allVideoAuctionEvents);
+events.addEvents(allVideoBidEvents);
+
+export function PbVideo(videoCore_, getConfig_, pbGlobal_, pbEvents_, videoEvents_, adServerCore_, videoImpressionVerifierFactory_) {
+  const videoCore = videoCore_;
+  const getConfig = getConfig_;
+  const pbGlobal = pbGlobal_;
+  const requestBids = pbGlobal.requestBids;
+  const pbEvents = pbEvents_;
+  const videoEvents = videoEvents_;
+  const adServerCore = adServerCore_;
+  const videoImpressionVerifierFactory = videoImpressionVerifierFactory_;
+  let videoImpressionVerifier;
+
+  function init() {
+    const cache = getConfig('cache');
+    videoImpressionVerifier = videoImpressionVerifierFactory(!!cache);
+    getConfig('video', ({ video }) => {
+      video.providers.forEach(provider => {
+        videoCore.registerProvider(provider);
+        videoCore.onEvents(videoEvents, (type, payload) => {
+          pbEvents.emit(type, payload);
+        }, provider.divId);
+
+        const adServerConfig = provider.adServer;
+        if (adServerConfig) {
+          adServerCore.registerAdServer(adServerConfig);
+        }
+      });
+    });
+
+    requestBids.before(enrichAdUnits, 40);
+
+    pbEvents.on(CONSTANTS.EVENTS.AUCTION_END, function(auctionResult) {
+      auctionResult.adUnits.forEach(adUnit => {
+        if (adUnit.video) {
+          renderWinningBid(adUnit);
+        }
+      });
+    });
+
+    pbEvents.on(CONSTANTS.EVENTS.BID_ADJUSTMENT, function (bid) {
+      videoImpressionVerifier.trackBid(bid);
+    });
+
+    pbEvents.on(AD_IMPRESSION, function (payload) {
+      triggerVideoBidEvent(BID_VIDEO_IMPRESSION, payload);
+    });
+
+    pbEvents.on(AD_ERROR, function (payload) {
+      triggerVideoBidEvent(BID_VIDEO_ERROR, payload);
+    });
+  }
+
+  return { init };
+
+  function enrichAdUnits(nextFn, bidRequest) {
+    const adUnits = bidRequest.adUnits || pbGlobal.adUnits || [];
+    adUnits.forEach(adUnit => {
+      enrichAdUnit(adUnit);
+    });
+    return nextFn.call(this, bidRequest);
+  }
+
+  function enrichAdUnit(adUnit) {
+    const videoMediaType = adUnit.mediaTypes.video;
+    const videoConfig = adUnit.video;
+    if (!videoMediaType || !videoConfig) {
+      return;
+    }
+
+    const oRtbParams = videoCore.getOrtbParams(videoConfig.divId);
+    if (!oRtbParams) {
+      return;
+    }
+
+    adUnit.mediaTypes.video = Object.assign({}, videoMediaType, oRtbParams.video);
+    let ortb2 = { ortb2: mergeDeep({}, getConfig('ortb2'), { site: oRtbParams.content }) };
+    pbGlobal.setConfig(ortb2);
+  }
+
+  function renderWinningBid(adUnit) {
```

Comment:
I am not sure about this - tagging it for other reviewers. I am concerned that this takes away control from the publisher, as it seems very different from what happens now with the result of an auction (which is just returned for them to do what they wish with it). 

Giving them a function that can render a bid seems more in line with the current approach.

Reply:
I understand that this goes against the pattern but the goal is to simplify the integration between prebid and video for publishers. I can bring this up next week at the Prebid Video sync to get the community's opinion.

Reply:
I agree with discussing it in the PMC. I have nothing against easy integration but I can also imagine publishers that can't use this because it has no way to accept their logic on which bids to render.

Reply:
addressed; `renderWinningBid` is only called if there is no bidsBackHandler and a render function is exposed